src/open-api/from-open-api.ts

```diff
@@ -53,7 +53,7 @@ export async function fromOpenApi(
       for (const baseUrl of serverUrls) {
         const path = normalizeSwaggerUrl(url)
         const requestUrl = isAbsoluteUrl(baseUrl)
-          ? new URL(path, baseUrl).href
```

Comment:
So, the subdirectory gets cut off because we've constructed the URL incorrectly. Specifcally, the `path` must always ends with a trailing slash. 

That would be the fix. Could you please adjust the pull request for that?